tuxedo/pulse/15/gen2/default.nix

```diff
@@ -0,0 +1,22 @@
+{
+  lib,
+  pkgs,
+  ...
+}: {
+  imports = [
+    ../../../../common/cpu/amd
+    ../../../../common/gpu/amd
+    ../../../../common/pc/laptop/ssd
+  ];
+
+  hardware.amdgpu.loadInInitrd = lib.mkDefault false;
+
+  services.udev.extraRules = builtins.concatStringsSep "\n" (
+    ["# Properly suspend the system."]
+    ++ (
+      map
+      (device: "SUBSYSTEM==\"pci\", ACTION==\"add\", ATTR{vendor}==\"0x144d\", ATTR{device}==\"${device}\", RUN+=\"${pkgs.bash}/bin/sh -c '${pkgs.coreutils}/bin/echo 0 | ${pkgs.coreutils}/bin/tee /sys/bus/pci/devices/$kernel/d3cold_allowed'\"")
```

Comment:
Should we do this?

```suggestion
      (device: "SUBSYSTEM==\"pci\", ACTION==\"add\", ATTR{vendor}==\"0x144d\", ATTR{device}==\"${device}\", RUN+=\"${pkgs.bash}/bin/sh -c 'echo 0 > /sys/bus/pci/devices/$kernel/d3cold_allowed'\"")
```

Reply:
Unlike `${pkgs.bash}/bin/sh`, `${pkgs.runtimeShell}` evaluates to `bash` (`/nix/store/<path>/bin/bash`):

```text
# Properly suspend the system.
SUBSYSTEM=="pci", ACTION=="add", ATTR{vendor}=="0x144d", ATTR{device}=="0xa80a", RUN+="/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash -c 'echo 0 > /sys/bus/pci/devices/$kernel/d3cold_allowed'"
SUBSYSTEM=="pci", ACTION=="add", ATTR{vendor}=="0x144d", ATTR{device}=="0xa808", RUN+="/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash -c 'echo 0 > /sys/bus/pci/devices/$kernel/d3cold_allowed'"
```

Would it not be better to just use `sh` (`${pkgs.bash}/bin/sh`)?

Reply:
Bash is fine. We use it anyway and the code is specific to nixos anyway.